### PR TITLE
Add support for LightGBM flavor

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Add support for lightgbm flavor.

## How is this patch tested?

Unit tests

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Model API now supports LightGBM.

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
The implementation is almost the same as XGBoost flavor.

**Comment by harupy** [comment]
ping @smurching

File: mlflow/lightgbm.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            "lightgbm=={}".format(lgb.__version__),
+        ],
+        additional_conda_channels=None)
+
+
+def save_model(lgb_model, path, conda_env=None, mlflow_model=Model()):
+    """
+    Save a LightGBM model to a path on the local file system.
+
+    :param lgb_model: LightGBM model to be saved.
```
Comment (smurching) [inline mlflow/lightgbm.py]:
```suggestion
    :param lgb_model: LightGBM model (an instance of lightgbm.Booster) to be saved.
```

Assuming my understanding is correct that the `save_model`, `log_model`, `load_model` APIs only work for `lightgbm.Booster` models, and not models that implement the [scikit-learn API](https://lightgbm.readthedocs.io/en/latest/Python-API.html#scikit-learn-api) - it'd also be good to add this clarification to the docstrings of `log_model` and `load_model`

**Review by smurching — COMMENTED** [review]
One small comment but otherwise looks good, thanks @harupy!

File: mlflow/lightgbm.py (hunk 1 above)
Comment (harupy) [reply mlflow/lightgbm.py]: @smurching Thanks for the comment. I'll add this clarification to XGBoost flavor as well.

File: mlflow/lightgbm.py [hunk 2]
```diff
@@ -158,8 +158,7 @@ def load_model(model_uri):
                       `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
                       artifact-locations>`_.
 
-    :return: An `lightgbm.Booster model object
```
Comment (smurching) [inline mlflow/lightgbm.py]: Ah, I think the old doc here was actually fine, sorry for the confusion - seems nice to link to the lightgbm docs, but this is also fine as is.

**Review by smurching — APPROVED** [review]
LGTM, looks like one of the TensorFlow tests is being flaky so retriggered it :) thanks @harupy!

**Comment by harupy** [comment]
@smurching Thanks!

File: mlflow/lightgbm.py (hunk 1 above)
Comment (harupy) [reply mlflow/lightgbm.py]: @smurching Do you think we should warn users that LightGBM and XGBoost flavor don't yet support models that implement the scikit-learn API?

Comment (smurching) [reply mlflow/lightgbm.py]: Yeah I think that'd definitely be helpful, I'd be happy to review a docstring update that adds such a warning, for example:

```suggestion
    :param lgb_model: LightGBM model (instance of `lightgbm.Booster <https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.Booster.html#lightgbm.Booster>`_) to be saved. Note that saving LightGBM models that implement the `scikit-learn API <https://lightgbm.readthedocs.io/en/latest/Python-API.html#scikit-learn-api>`_  is not supported.
```

Comment (harupy) [reply mlflow/lightgbm.py]: @smurching Thanks. I'll open a PR to add this warning.

Comment (harupy) [reply mlflow/lightgbm.py]: @smurching I opened a PR to update docstring https://github.com/mlflow/mlflow/pull/2170

**Pull request merged**